### ISSUE TITLE
MD013 tables=false not applied to indented tables inside list items

### ISSUE THREAD

**Issue by scottamain** [rvben/rumdl#636]
With MD060 (table column alignment) enabled and `tables = false` configured for MD013 (via `[line-length]` or `[MD013]`), rumdl correctly skips line-length checks for **top-level** Markdown tables. However, it still reports `[MD013]` for **indented** table rows nested inside a list item, even when those rows exceed 80 characters only because of column alignment.

We rely on `MD013.tables = false` plus [MD060's documented behavior](https://github.com/rvben/rumdl/blob/main/docs/md060.md) that when `MD013.tables = false`, `max-width` is unlimited and tables are not auto-compacted for line length. The bug breaks that combination for indented tables inside lists.

There is also a secondary problem: the reported line number often does not match the long physical line (see examples below).

## Configuration

Minimal repro (MD013 only — bug still reproduces):

```toml
[line-length]
line-length = 80
tables = false

[MD013]
reflow = true
```

Configuration matching our project (MD060 explicitly enabled):

```toml
[line-length]
line-length = 80
tables = false

[MD013]
reflow = true

# MD060 is opt-in (enabled = false by default). We enable it for aligned tables.
[MD060]
enabled = true
style = "aligned-no-space"
```

With `tables = false`, [MD060's docs](https://github.com/rvben/rumdl/blob/main/docs/md060.md) state that `max-width = 0` inherits **unlimited** width (no auto-compact). Wide aligned rows are intentional; MD013 should not flag them. That works for top-level tables but not for indented tables in list items.

## Version

```
rumdl 0.1.62
```

## Steps to reproduce

1. Save one of the configs above as `.rumdl.toml`.
2. Create the sample files below.
3. Run:

   ```bash
   rumdl check unindented-table.md indented-table-in-list.md --config .rumdl.toml
   ```

## Sample: `unindented-table.md` (control — passes)

```markdown
# Unindented table (control)

| Removed                                 | Replacement                              |
|-----------------------------------------|------------------------------------------|
| `get_type_name[T]()`                    | `reflect[T].name()`                      |
```

**Expected:** No MD013 violations (table rows are 88 characters; `tables = false` applies).

**Actual:** No issues. ✅

## Sample: `indented-table-in-list.md` (fails)

```markdown
# Indented table inside list item

- Intro paragraph before the table.

  | Removed                                 | Replacement                              |
  |-----------------------------------------|------------------------------------------|
  | `get_type_name[T]()`                    | `reflect[T].name()`                      |

- Next list item.
```

**Expected:** No MD013 violations (same table content as control; still a table; `tables = false`).

**Actual:**

```
indented-table-in-list.md:3:1: [MD013] Line length 88 exceeds 80 characters [*]
```

- Line 3 is `- Intro paragraph before the table.` (31 characters).
- The 88-character lines are physical lines 5–7 (indented table rows).

## Sample: `misleading-line-number.md` (fails; line number mismatch)

```markdown
# Misleading line number in larger document

## Section

- `List[T]` no longer requires its type to be `Copyable`, but now works with
  `Movable`-only types. Iteration still requires `Copyable` and will emit
  a `comptime assert` if not satisfied.

- `reflect[T]` is now a `comptime` alias for the `Reflected[T]` handle
  type rather than a function returning a zero-sized handle instance.

  removed; use the corresponding methods on `reflect[T]`:

  | Removed                                 | Replacement                              |
  |-----------------------------------------|------------------------------------------|
  | `get_type_name[T]()`                    | `reflect[T].name()`                      |

- Added `ReflectedFn[func]`, a function-side reflection handle.
```

**Actual:**

```
misleading-line-number.md:9:1: [MD013] Line length 88 exceeds 80 characters [*]
```

- Line 9 is 70 characters (`- `reflect[T]` is now a ... handle`).
- The 88-character lines are physical lines 14–16 (indented table rows).

## Workaround

Dedent the table to column 0 (ends the list item block in many renderers), disable MD013 around the table with inline `<!-- rumdl-disable MD013 -->` comments, or use `[per-file-ignores]`.

## Suggested fix

Treat indented pipe tables inside list items as tables for MD013 purposes when `tables = false`, consistent with top-level tables. Report diagnostics on the actual overflowing table row line/column.

**Comment by scottamain** [comment]
Oops. This was fixed recently and I just need to update. Thank you!
https://github.com/rvben/rumdl/commit/fc3f89de
